src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -429,9 +429,12 @@ int MPII_Genutil_sched_start(MPII_Genutil_sched_t * sched, MPIR_Comm * comm, MPI
 
     /* Enqueue schedule and activate progress hook if not already activated */
     reqp->u.nbc.coll.sched = (void *) sched;
-    if (coll_queue.head == NULL)
-        MPID_Progress_activate_hook(MPII_Genutil_progress_hook_id);
+
+    MPID_THREAD_CS_ENTER(VCI, MPIDIU_THREAD_TSP_QUEUE_MUTEX);
     DL_APPEND(coll_queue.head, &(reqp->u.nbc.coll));
+    MPID_THREAD_CS_EXIT(VCI, MPIDIU_THREAD_TSP_QUEUE_MUTEX);
+
+    MPID_Progress_activate_hook(MPII_Genutil_progress_hook_id);
```

Comment:
Same here. A comment about why we removed the check for `coll_queue.head == NULL` in the commit message would be helpful.